### ISSUE TITLE
Palworld - Query returning state after server is shut down

### ISSUE THREAD

**Issue by Cabecinha84** [gamedig/node-gamedig#518]
How to reproduce:
- Start your Palworld server;
- Execute 'gamedig --type palworld ipforyourserver:8211' and you get a state object;
- Shutdown your Palworld server;
- Execute 'gamedig --type palworld ipforyourserver:8211' and you get a state object;

It takes several after the server is no longer running for returning the expected error result:
"error":"Failed all 2 attempts".

**Comment by podrivo** [comment]
This could be due to the server information not having enough time to be propagated or maybe a cache config?
It doesn't seem to be gamedig related, since the query is working as expected.

**Comment by Cabecinha84** [comment]
> This could be due to the server information not having enough time to be propagated or maybe a cache config? It doesn't seem to be gamedig related, since the query is working as expected.

Yes, seems like cache but from gamedig, this is gamedig specific.
The server is stopped, I can't connect to the game server, and gamedig query continues to return the state object for several minutes, and I have tested the gamedig from different servers.
But I can't find any reference on gamedig docs about cache or how to clear it/not use it.

Any help with this would be appreciated.
Thank you.

**Comment by CosminPerRam** [comment]
> But I can't find any reference on gamedig docs about cache or how to clear it/not use it.

GameDig doesnt cache responses.
Palworld uses Epic Online Services, the library does a query to it to get your server information, this is a problem regarding EOS, not gamedig, as after you close your server, you'd expect for it to be gone, but it's still remaining there for a little while, so gamedig does a query and its still there (it shouldnt).

**Comment by podrivo** [comment]
> The server is stopped, I can't connect to the game server, and gamedig query continues to return the state object for several minutes, and I have tested the gamedig query from different servers and always got the same behaviour.

This is probably due to some server config, that lets gamedig query a response, even though you can't connect to it.
You should look for something on how Epic Online Services cache their server info.

**Comment by Cabecinha84** [comment]
> > The server is stopped, I can't connect to the game server, and gamedig query continues to return the state object for several minutes, and I have tested the gamedig query from different servers and always got the same behaviour.
> 
> This is probably due to some server config, that lets gamedig query a response, even though you can't connect to it. You should look for something on how Epic Online Services cache their server info.

Server config, where the Palword server is running? How can it be a server config, if the server is shut down?
So gamedig don't connect to the server host/port I ask to query? Calls another service to get the response? And it can be a cache on that service?
Trying to understand how gamedig works.

Thank you for all your input so far.

**Comment by podrivo** [comment]
Yes, where Palworld is running. It does not call another service.

As I mentioned, server might be offline to players to connect, but might have a cache config that allows gamedig to still query it. I don't have more information on how Palworld servers work, but that seems to be the case.

**Comment by GuilhermeWerner** [comment]
No, you are confusing, the servers that use EOS (Epic Online Services), such as ARK: Survival Ascended, Palworld, etc. do not respond directly to queries like other games that use the valve protocol, so we can obtain the data, a request is sent to Epic's web api, with the game's credentials (so that gamedig is treated as the game's client) and the server's IP, in order to get the query response.

In other words, this cache is due to Epic's API, and is outside the control of the server administrator and gamedig.

**Comment by Cabecinha84** [comment]
> No, you are confusing, the servers that use EOS (Epic Online Services), such as ARK: Survival Ascended, Palworld, etc. do not respond directly to queries like other games that use the valve protocol, so we can obtain the data sending a request to Epic's web api, with the game's credentials (so that gamedig is treated as the game's client) and the server's IP, to then get the query response.
> 
> In other words, this cache is due to Epic's API, and is outside the control of the server administrator and gamedig.

Now makes more sense. I will close the bug.
Thank you for your explanation.

**Comment by CosminPerRam** [comment]
I thought about these games if we could 'ping' them, but I don't really know how reliable would that be and if possible at all.
